doc/subsystems/bluetooth/hci.txt

```diff
@@ -0,0 +1,994 @@
+	|       | 2    | Read Supported Features                    |
+	|       | 3    | Set Event Mask                             |
+	|       | 4    | Reset                                      |
+	|       | 5    | Write BD_ADDR                              |
+	|       | 6    | Set Trace Enable                           |
+	|       | 7    | Read Build Information                     |
+	+-------+------+--------------------------------------------+
+	| 1     | 0    | Read Static Addresses                      |
+	|       | 1    | Read Key Hierarchy Roots                   |
+	|       | 2    | Read Chip Temperature                      |
+	|       | 3    | Read Host Stack Commands                   |
+	|       | 4    | Set Scan Request Reports                   |
+	|       | 5    | Reserved                                   |
+	|       | 6    | Reserved                                   |
+	|       | 7    | Reserved                                   |
+	+-------+------+--------------------------------------------+
+
+Only Read_Version_Information and Read_Supported_Commands commands are
+mandatory.
+
+
+Zephyr Vendor Supported Features
+================================
+
+The Supported Features configuration parameter lists vendor specific features
+that either extend commands or don't have a vendor command. Most features are
+indicated by the support its respective vendor command. The Supported Features
+are represented as an 8 octet bit mask. For each feature a single bit is
+specified which shall be set to 1 if the feature is supported and set to 0
+otherwise.
+
+	+-------+------+--------------------------------------------+
+	| Octet | Bit  | Feature Supported                          |
+	+-------+------+--------------------------------------------+
+	| 0     | 0    | Vendor Diagnostic Channel                  |
+	|       | 1    | Reserved                                   |
+	|       | 2    | Reserved                                   |
+	|       | 3    | Reserved                                   |
+	|       | 4    | Reserved                                   |
+	|       | 5    | Reserved                                   |
+	|       | 6    | Reserved                                   |
+	|       | 7    | Reserved                                   |
+	+-------+------+--------------------------------------------+
+
+
+Zephyr Vendor Event Mask
+========================
+
+The Event Mask configuration parameter lists the events that are indicated by
+the controller. By default no vendor events are reported.
+
+	+-------+------+--------------------------------------------+
+	| Octet | Bit  | Event                                      |
+	+-------+------+--------------------------------------------+
+	| 0     | 0    | Reserved                                   |
+	|       | 1    | Fatal Error                                |
+	|       | 2    | Trace Information Event                    |
+	|       | 3    | Scan Request Received Event                |
```

Comment:
Yes, it is already part of Extended Advertising HCI commands. That description makes it clear that this is only for legacy advertising set. So in case you need to safe code space and do not require any of the extended advertising features, you can use this. It is used as of today, so it seems like a good addition. Also all vendor features are optional. We can hide this behind a config option if only extended advertising commands are used.